src/dbus/error.cpp

```diff
@@ -0,0 +1,119 @@
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "dbus/error.hpp"
+
+#include "dbus/constants.hpp"
+#include "dbus/dbus_message_helper.hpp"
+
+static const std::pair<otError, const char *> sErrorNameMap[] = {
+    {OT_ERROR_GENERIC, "io.openthread.Error.Generic"},
+    {OT_ERROR_NONE, "io.openthread.Error.OK"},
+    {OT_ERROR_FAILED, "io.openthread.Error.Failed"},
+    {OT_ERROR_DROP, "io.openthread.Error.Drop"},
+    {OT_ERROR_NO_BUFS, "io.openthread.Error.NoBufs"},
+    {OT_ERROR_NO_ROUTE, "io.openthread.Error.NoRoute"},
+    {OT_ERROR_BUSY, "io.openthread.Error.Busy"},
+    {OT_ERROR_PARSE, "io.openthread.Error.Parse"},
+    {OT_ERROR_INVALID_ARGS, "io.openthread.Error.InvalidArgs"},
+    {OT_ERROR_SECURITY, "io.openthread.Error.Security"},
+    {OT_ERROR_ADDRESS_QUERY, "io.openthread.Error.AddressQuery"},
+    {OT_ERROR_NO_ADDRESS, "io.openthread.Error.NoAddress"},
+    {OT_ERROR_ABORT, "io.openthread.Error.Abort"},
+    {OT_ERROR_NOT_IMPLEMENTED, "io.openthread.Error.NotImplemented"},
+    {OT_ERROR_INVALID_STATE, "io.openthread.Error.InvalidState"},
+    {OT_ERROR_NO_ACK, "io.openthread.Error.NoAck"},
+    {OT_ERROR_CHANNEL_ACCESS_FAILURE, "io.openthread.Error.ChannelAccessFailure"},
+    {OT_ERROR_DETACHED, "io.openthread.Error.Detached"},
+    {OT_ERROR_FCS, "io.openthread.Error.FcsErr"},
+    {OT_ERROR_NO_FRAME_RECEIVED, "io.openthread.Error.NoFrameReceived"},
+    {OT_ERROR_UNKNOWN_NEIGHBOR, "io.openthread.Error.UnknownNeighbor"},
+    {OT_ERROR_INVALID_SOURCE_ADDRESS, "io.openthread.Error.InvalidSourceAddress"},
+    {OT_ERROR_ADDRESS_FILTERED, "io.openthread.Error.AddressFiltered"},
+    {OT_ERROR_DESTINATION_ADDRESS_FILTERED, "io.openthread.Error.DestinationAddressFiltered"},
+    {OT_ERROR_NOT_FOUND, "io.openthread.Error.NotFound"},
+    {OT_ERROR_ALREADY, "io.openthread.Error.Already"},
+    {OT_ERROR_IP6_ADDRESS_CREATION_FAILURE, "io.openthread.Error.Ipv6AddressCreationFailure"},
+    {OT_ERROR_NOT_CAPABLE, "io.openthread.Error.NotCapable"},
+    {OT_ERROR_RESPONSE_TIMEOUT, "io.openthread.Error.ResponseTimeout"},
+    {OT_ERROR_DUPLICATED, "io.openthread.Error.Duplicated"},
+    {OT_ERROR_REASSEMBLY_TIMEOUT, "io.openthread.Error.ReassemblyTimeout"},
+    {OT_ERROR_NOT_TMF, "io.openthread.Error.NotTmf"},
+    {OT_ERROR_NOT_LOWPAN_DATA_FRAME, "io.openthread.Error.NonLowpanDatatFrame"},
+    {OT_ERROR_LINK_MARGIN_LOW, "io.openthread.Error.LinkMarginLow"},
+};
+
+namespace otbr {
+namespace dbus {
+
+const char *ConvertToDBusErrorName(otError aError)
+{
+    const char *name = sErrorNameMap[0].second;
+
+    for (auto &&p : sErrorNameMap)
+    {
+        if (p.first == aError)
+        {
+            name = p.second;
+        }
+    }
+    return name;
+}
+
+otError ConvertFromDBusErrorName(const std::string &aErrorName)
```

Comment:
Why don't we use `const char *` to keep the signature consistent with `ConvertToDBusErrorName`?

Reply:
We use std::string here to support both const char* and std::string and to avoid the strcmp memory issue (you cannot really decide the length of a const char * that is safe to read).